Minor refactor of number-to-string conversion

peter-b opened the pull request:
Small refactor of number-to-string conversion using `MCU_r8tos()`:
- Fix a buffer that was being allocated with `new[]` and destroyed with `delete`
- Always set the numeric value when creating a stringref
- Remove duplicated code from various `MCU_r8tos()` consumers that subsequently construct a string ref, such as `MCExecContext::ConvertToString()`

livecodeali commented on engine/src/util.cpp:
```diff
@@ -593,20 +593,23 @@ bool MCU_r8tos(real8 n, uint2 fw, uint2 trailing, uint2 force, MCStringRef &r_st
+
+	delete[] t_str;
+
 	return true;
```
Should be `return t_success`

peter-b commented:
Per conversation with @livecodeali, this could do with a more thorough refactor.

peter-b committed:
MCU_r8tos(): Set numeric value when creating a string.

peter-b committed:
MCExecContext::ConvertToString(): Use MCU_r8tos() to create stringref

peter-b committed:
MCValueConvertToStringForSave(): Use MCU_r8tos() to create stringref

peter-b committed:
MCStringCreateWithCStringAndRelease(): Release with delete[].
    
I'm not sure if this is correct, but most of its callers allocate the
buffer passed in with `new[]` so let's go with that for now.

peter-b committed:
MCExtensionTryToConvertFromScriptType(): Use MCU_r8tos() to create stringref

peter-b commented:
@livecodeali I've now updated this PR with the more extensive refactor you suggested.

peter-b committed:
Don't use auto keyword yet.

livecodeali commented:
@livecode-vulcan review ok e6c6382

livecode-vulcan commented:
💙 review by @livecodeali ok e6c63822eba6ce81b675f4920e7741c5545c49b5

livecode-vulcan commented:
😎 test success e6c63822eba6ce81b675f4920e7741c5545c49b5
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2793)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2802)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2788)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2804)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2810)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2814)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2833)
